src/particles/transformation/TO_FIXED_S.H

```diff
@@ -57,25 +60,37 @@ namespace transformation
             amrex::ParticleReal const y = p.pos(1);
             amrex::ParticleReal const t = p.pos(2);
 
-            // arguments for ptd and pt
+            // compute value of reference ptd = -gamma
             amrex::ParticleReal const argd = 1.0_prt + pow(m_pzd, 2);
-            amrex::ParticleReal const arg = 1.0_prt + pow(m_pzd+pt, 2) + pow(px, 2) + pow(py, 2);
-
-            // check for validity of arguments
             AMREX_ALWAYS_ASSERT_WITH_MESSAGE(argd > 0.0_prt, "invalid ptd arg (<=0)");
-            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(arg > 0.0_prt, "invalid pt arg (<=0)");
-
-            // transform position and momentum (t2z)
-            //   (universal for all elements - needs tracking for RF elements)
             amrex::ParticleReal const ptdf = argd > 0.0_prt ? -sqrt(argd) : -1.0_prt;
+
+            // transform momenta to dynamic units (eg, so that momenta are
+            // normalized by mc):
+            px = px*m_pzd;
+            py = py*m_pzd;
+            pt = pt*m_pzd;
+
+            // compute value of particle pt = -gamma
+            amrex::ParticleReal const arg = 1.0_prt + pow(m_pzd+pt, 2) + pow(px, 2) + pow(py, 2);
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(arg > 0.0_prt, "invalid pt arg (<=0)");
```

Comment:
Move to debug-only because this is in a kernel as part of a hot loop.
```suggestion
            AMREX_ASSERT_WITH_MESSAGE(arg > 0.0_prt, "invalid pt arg (<=0)");
```
https://github.com/ECP-WarpX/impactx/pull/172#discussion_r961186464